zerver/tests/test_channel_permissions.py

```diff
@@ -887,6 +882,24 @@ def check_channel_property_update(user: UserProfile, error_msg: str | None = Non
         check_channel_property_update(self.guest, error_msg=default_error_msg)
         self.unsubscribe(self.guest, stream.name)
 
+        # Check for permission in unsubscribed private streams.
```

Comment:
I think it is fine to not check all types of groups here and we can avoid checking `subscribed_to_stream` for all cases.